src/main/java/com/google/sps/servlets/CareerQuizServlet.java

```diff
@@ -17,21 +19,44 @@
 /** Responds with a JSON string containing questions and answers for the career quiz*/
 public class CareerQuizServlet extends HttpServlet {
   private final CareerQuestionDatabase careerQuestionDatabase;
-  private static final String JSON_CONTENT_TYPE = "application/json;";
+  private ArrayList<CareerQuestionAndChoices> questionsAndChoices;
+  private static final String JSON_CONTENT_TYPE = "application/json";
+  private static final String QUIZ_SUBMIT = "career-quiz-submit";
+  private static final Gson gson = new Gson();
   
   @Inject public CareerQuizServlet(CareerQuestionDatabase careerQuestionDatabase) {
      this.careerQuestionDatabase = careerQuestionDatabase;
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    ArrayList<CareerQuestionAndChoices> questionsAndChoices = 
-        this.careerQuestionDatabase.getQuestionsAndChoices();
+    this.questionsAndChoices = this.careerQuestionDatabase.getQuestionsAndChoices();
     // Convert the ArrayLists to JSON
     Gson gson = new Gson();
     String questionsAndChoicesJson = gson.toJson(questionsAndChoices);
-    // Send the JSON as the response
-    response.setContentType(JSON_CONTENT_TYPE);
+    // Send the career questions and choices JSON as the response    response.setContentType(JSON_CONTENT_TYPE);
     response.getWriter().println(questionsAndChoicesJson);
   }
+
+  @Override
+  /** when submit button on career quiz is clicked, this method is called and returns recommended path
+  based on selected choices*/
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    this.questionsAndChoices = this.careerQuestionDatabase.getQuestionsAndChoices();
+    String recommendedPath = handleQuizSubmission(request);
+    response.setContentType(JSON_CONTENT_TYPE);
+    String recommendedPathJson = gson.toJson(recommendedPath);
+    response.getWriter().println(recommendedPathJson);
```

Comment:
You can inline this `response.getWriter().println(gson.toJson(recommendedPath));`

Reply:
done.